### ISSUE TITLE
Error when installing/enabling extension

### ISSUE THREAD

**Issue by AzP** [eonpatapon/gnome-shell-extensions-mediaplayer#227]
When I enable the extension from https://extensions.gnome.org/extension/55/media-player-indicator/ I get an error. It has been like this ever since I switched to Gnome 3.16 (I think). I'm now on 3.18. This is the log from systemd:

```
Nov 26 23:22:57 skare gnome-session[698]: Archive:  /tmp/OOMN8X.shell-extension.zip
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/nl/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pl/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_TW/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_CN/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/tr/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/schemas/org.gnome.shell.extensions.mediaplayer.gschema.xml
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/widget.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/de/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/it/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ro/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/prefs.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/settings.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/lt/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/stylesheet.css
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/it/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_CN/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/dbus.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/extension.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pt_BR/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/es/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/player.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ro/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sk/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sr/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/tr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/ui.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/panel.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/nl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/gl/
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/manager.js
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/fr/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sk/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/gl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/schemas/gschemas.compiled
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/COPYING
Nov 26 23:22:57 skare gnome-session[698]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/LC_MESSAGES/
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/fr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/lib.js
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/es/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
Nov 26 23:22:57 skare gnome-session[698]: extracting: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/metadata.json
Nov 26 23:22:57 skare gnome-session[698]: Gjs-Message: JS LOG: Extension "mediaplayer@patapon.info" had error: TypeError: this.app is null
Nov 26 23:22:57 skare gnome-session[698]: (gnome-shell:806): GLib-CRITICAL **: g_variant_type_checked_: assertion 'g_variant_type_string_is_valid (type_string)' failed
```

**Comment by eonpatapon** [comment]
Do you have a default application for music files ?

You can check this in gnome-control-center > Details > Default Applications

**Comment by AzP** [comment]
I'm not sure, I'll check it out. I mainly use Spotify but also rhythmbox every now and then.

**Comment by AzP** [comment]
Yes, I had rhythmbox as default application for music files.

**Comment by AzP** [comment]
I tried uninstalling it, making sure rhythmbox is the default application set in Gnome, restarted Gnome shell (so it wasn't running with the extension, and then tried installing it again:

```
dec 12 12:45:36 skare gnome-session[743]: Archive:  /tmp/8ZVE9X.shell-extension.zip
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/nl/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pl/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_TW/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_CN/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/tr/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/schemas/org.gnome.shell.extensions.mediaplayer.gschema.xml
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/widget.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/de/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/it/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ro/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/prefs.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/settings.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/lt/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/stylesheet.css
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/it/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/zh_CN/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/dbus.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/extension.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pt_BR/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/es/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/player.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ro/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sk/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sr/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/tr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/ui.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/panel.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/nl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/pl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/gl/
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/he/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/manager.js
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/fr/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sk/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/gl/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/schemas/gschemas.compiled
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/sr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/COPYING
dec 12 12:45:36 skare gnome-session[743]: creating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/LC_MESSAGES/
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/fr/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/lib.js
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/es/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: inflating: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/locale/ru/LC_MESSAGES/gnome-shell-extensions-mediaplayer.mo
dec 12 12:45:36 skare gnome-session[743]: extracting: /home/peter/.local/share/gnome-shell/extensions/mediaplayer@patapon.info/metadata.json
dec 12 12:45:36 skare gnome-session[743]: Gjs-Message: JS LOG: Extension "mediaplayer@patapon.info" had error: TypeError: this.app is null
dec 12 12:47:18 skare gnome-session[743]: (gnome-shell:847): GLib-CRITICAL **: Source ID 134 was not found when attempting to remove it
```

**Comment by AzP** [comment]
I forgot to mention, that it still ends up as one of my installed extensions, and it does show up in the dropdown menu in Gnome.
![screenshot from 2015-12-12 12-53-51](https://cloud.githubusercontent.com/assets/954714/11761757/97452fb0-a0cf-11e5-80af-0e191e1d15d7.png)

**Comment by eonpatapon** [comment]
Ok.

What if you disable the setting "Allow to start the default media player" in the extension preferences and then you disable/enable the extension.

Do you get the error message ?

**Comment by AzP** [comment]
That "fixes" the problem! No error message, and extension is enabled successfully.

**Comment by eonpatapon** [comment]
Maybe fixed by #295

**Comment by JasonLG1979** [comment]
@AzP did #295 fix this issue for you?
